### PR TITLE
Minor fixes in stm32mp1 and its drivers

### PR DESCRIPTION
**Pull request by etienne-lms**
This series bring minor fixes in the platform and its drivers:
- use `vaddr_t` instead of `uintptr_t` when manipulating a physical or virtual address,
- initialize each local variables where its is defined,
- rename local variable `excep` into pore explicit `exceptions`,
- add few `__maybe_unused` attributes where applicable.

### CODE REVIEW

**Comment by jenswi-linaro** [comment]
Please fix the checkpatch error.

File: core/arch/arm/plat-stm32mp1/main.c [hunk 1]
```diff
@@ -233,24 +233,18 @@ static TEE_Result init_late_stm32mp1_drivers(void)
...
-
-	return pbase;
+	return io_pa_or_va(&base);
 }
 
-uintptr_t get_gicd_base(void)
+vaddr_t get_gicd_base(void)
 {
-	uintptr_t pbase = GIC_BASE + GICD_OFFSET;
+	struct io_pa_va base = { .pa = GIC_BASE + GICD_OFFSET };
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-stm32mp1/main.c]: `static`?

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c [hunk 2]
```diff
@@ -500,12 +500,12 @@ static enum stm32mp1_parent_id stm32mp1_clk_get_fixed_parent(int i)
...
-	unsigned int j;
-	uint32_t p_sel;
-	int i;
-	enum stm32mp1_parent_id p;
-	enum stm32mp1_parent_sel s;
+	const struct stm32mp1_clk_sel *sel = NULL;
+	unsigned int j = 0;
+	uint32_t p_sel = 0;
+	int i = 0;
+	enum stm32mp1_parent_id p = (enum stm32mp1_parent_id)0;
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c]: ok.

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c [hunk 3]
```diff
@@ -963,19 +968,19 @@ static unsigned int clk_freq_prop(void *fdt, int node)
 
 static void get_osc_freq_from_dt(void *fdt)
 {
-	enum stm32mp_osc_id idx;
+	enum stm32mp_osc_id idx = (enum stm32mp_osc_id)0;
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c]: ok.

File: core/arch/arm/plat-stm32mp1/main.c [hunk 4]
```diff
@@ -233,24 +233,18 @@ static TEE_Result init_late_stm32mp1_drivers(void)
 }
 driver_init_late(init_late_stm32mp1_drivers);
 
-uintptr_t get_gicc_base(void)
+vaddr_t get_gicc_base(void)
 {
-	uintptr_t pbase = GIC_BASE + GICC_OFFSET;
+	struct io_pa_va base = { .pa = GIC_BASE + GICC_OFFSET };
```
[the remark this answers is not in the document]
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/main.c]: This is seldom used, I did not use a static to consume stack rather than heap.

File: core/arch/arm/plat-stm32mp1/main.c (hunk 1 above)
Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/main.c]: ditto.

**Comment by etienne-lms** [comment]
ping

File: core/drivers/stm32_etzpc.c [hunk 5]
```diff
@@ -294,7 +294,7 @@ static void get_hwcfg(struct etzpc_hwcfg *hwcfg)
 static void init_devive_from_hw_config(struct etzpc_instance *dev,
 					      paddr_t pbase)
 {
-	struct etzpc_hwcfg hwcfg;
+	struct etzpc_hwcfg hwcfg = { 0 };
```
Comment (jforissier) [inline core/drivers/stm32_etzpc.c]: `memset()` for this one since it's a struct?

Comment (jforissier) [reply core/drivers/stm32_etzpc.c]: > `memset()` for this one since it's a struct?

Actually, I'm trying to remember what we have decided for structs. At some point we said `memset()` to zero is preferable over `= { 0 };`, and I think there were two reasons:
1. No need to worry about `= { 0 };` vs. `= { NULL };` anymore. Weak argument, because GCC has `= { };`.
2. Padding is initialized by `memset()`, whereas it is left undefined by ` = { ... };` which will make a difference if structures are compared using `memcmp()`. Not a very strong argument either IMO, because we normally do not do that, and if we did, then initializing with valid values with `= { val };` would not be OK and would need to zero the memory first.

So... what should this line be? My preference goes to `struct etzpc_hwcfg hwcfg = { };` but perhaps I missed something?

Comment (jenswi-linaro) [reply core/drivers/stm32_etzpc.c]: A problem with `= { };` (if I recall correctly) is that it can cause warnings with old GCC versions.

I agree that case 2 is not relevant for OP-TEE. Doing the `memset()` thing is ugly and somewhat clumsy. So I'm in favor of finding something else.

Comment (jforissier) [reply core/drivers/stm32_etzpc.c]: > A problem with = { }; (if I recall correctly) is that it can cause warnings with old GCC versions.

Hopefully we don't care about these older versions?

Comment (jenswi-linaro) [reply core/drivers/stm32_etzpc.c]: It seems that `= { }` isn't valid C code even if it's accepted by gcc (and clang?).
https://stackoverflow.com/questions/17589533/is-an-empty-initializer-list-valid-c-code

The universal zero initializer is `= { 0 }` in C (but `= { }` in C++). There's a gcc bug that relates to this.
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53119

However, that bug is quite old and the warning can be silenced with with -Wno-missing-braces

Comment (etienne-lms) [reply core/drivers/stm32_etzpc.c]: I would suggest to use `memset()` in this P-R for consistency with other core sources, and eventually move all those `memset()` to an alternate way once we end up with some agreement with how to handle all those inits.

Maybe at least all `enum` types initialization could move to the C universal `{ 0 }` . As for the structures, I really think `struct foo bar = <something>` is cleaner then the `memset()` way.

Comment (jforissier) [reply core/drivers/stm32_etzpc.c]: I still think `= { };` is the most convenient. It's used all over the place in Linux.

> I would suggest to use memset() in this P-R for consistency with other core sources, [...]

The patch is about initializing variables that objectively don't need to be initialized. This falls in the "coding rules" category. If we don't agree on the rules, what sense can this patch make? ;-)

Comment (etienne-lms) [reply core/drivers/stm32_etzpc.c]: Sorry, i got it wrong. optee_os has not yet moved to init all local variables, only optee_client did (https://github.com/OP-TEE/optee_client/pull/151) and optee_test is in progress (https://github.com/OP-TEE/optee_test/pull/339 & https://github.com/OP-TEE/optee_test/pull/340).

Comment (jenswi-linaro) [reply core/drivers/stm32_etzpc.c]: OK, let's use `= { };` then.

File: core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c [hunk 6]
```diff
@@ -963,19 +968,20 @@ static unsigned int clk_freq_prop(void *fdt, int node)
...
 {
-	enum stm32mp_osc_id idx;
+	enum stm32mp_osc_id idx = _UNKNOWN_OSC_ID;
 	int clk_node = fdt_path_offset(fdt, "/clocks");
 
 	if (clk_node < 0)
 		panic();
 
-	for (idx = (enum stm32mp_osc_id)0 ; idx < NB_OSC; idx++) {
+	assert((int)_HSI == 0);
```
Comment (jenswi-linaro) [inline core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c]: Why is this needed? Consider using `COMPILE_TIME_ASSERT()` if it's needed at all.

Comment (etienne-lms) [reply core/arch/arm/plat-stm32mp1/drivers/stm32mp1_clk.c]: True, thanks!
I'll use `COMPILE_TIME_ASSERT()`. I think it is preferable for readers to understand that this loop expects to cover all the relevant IDs of this enumerated type.

**Comment by etienne-lms** [comment]
comment addressed. fixup commits squashed. tags applied.
thanks.

**Comment by etienne-lms** [comment]
rebased to fix merge conflict.
